### PR TITLE
SAT: unit tests `test_state_with_abnormally_large_values`  + add missing `future_state_path`

### LINKED ISSUE #17338: SAT verifies that an incremental stream with no data still produces a state message
After an OC issue, we added this to the CDK - https://github.com/airbytehq/airbyte/pull/17296.  However, for connectors which don't use the CDK we should check this assertion

```
In an incremntal sync
if the cursor is already at the highwatermark
(records might be emitted)
but the same state message as before should be re-emitted in the sync
```

We already have a test for "abnormal state" - start looking here

This story is not to make all 200 connectors pass this test, but probably test with one or 2 connectors

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte/issues/17338
Just a sanity check over our "source should always emit state" testing.

## How
* Unit test the `test_state_with_abnormally_large_values` to make sure it works as expected
* Add missing `future_state_path` to `acceptance-test-config.yml` of the following connectors:
  * source-surveymonkey [GA]
  * source-bigcommerce [alpha]
  * source-talkdesk-explore [alpha]
  * source-paystack [alpha]

The other connector missing `future_state_path` testing are the following:
* source-amplitude [GA]
* source-snapchat-marketing [GA]
* source-zendesk-chat [GA]
* source-aws-cloudtrail [alpha]
* source-strava [alpha]

It's not possible to check these connector always emit a state when no record is available because their API does not allow querying future data, which is what `test_state_with_abnormally_large_values` is testing.

### CODE REVIEW

**Comment by alafanechere** [comment]
I don't expect SAT to pass on alpha connectors are they are also failing on master...

File: airbyte-integrations/connectors/source-bigcommerce/acceptance-test-config.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ tests:
   incremental:
     - config_path: "secrets/config.json"
       configured_catalog_path: "integration_tests/configured_catalog.json"
-      # future_state_path: "integration_tests/abnormal_state.json"
+      future_state_path: "integration_tests/abnormal_state.json"
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-bigcommerce/acceptance-test-config.yml:21]: I love to see more tests being enabled!

**Review by evantahler — APPROVED** [review]
👍 

A question about "The other connector missing future_state_path testing are the following" 
- did you audit all the connectors?  How did you come up with this list?

A question about: "I don't expect SAT to pass on alpha connectors are they are also failing on master..."
* How did you learn this?  I wonder if we should keep a spreadsheet or metabase report about this to help connector developers know what to do if they need to re-publish a connector

**Comment by alafanechere** [comment]
> did you audit all the connectors? How did you come up with this list?

Quick script parsing the `acceptance-test-config.yml` files of each connector declaring `incremental` test. Declared missing if `future_state_path` is not set.
I'd like to suggest a permanent approach at this coverage evaluation in the tech spec of https://github.com/airbytehq/airbyte/issues/17531

> A question about: "I don't expect SAT to pass on alpha connectors are they are also failing on master..."
How did you learn this? I wonder if we should keep a spreadsheet or metabase report about this to help connector developers know what to do if they need to re-publish a connector

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md

**Comment by alafanechere** [comment]
/publish connector=bases/source-acceptance-test auto-bump-version=false

**Comment by alafanechere** [comment]
/publish connector=bases/source-acceptance-test auto-bump-version=false

**Comment by alafanechere** [comment]
/publish connector=bases/source-acceptance-test auto-bump-version=false

**Comment by alafanechere** [comment]
/publish connector=bases/source-acceptance-test auto-bump-version=false

**Pull request merged**